File: python/71-80/Sort Colors.py
    # @param A a list of integers
    # @return nothing, sort in place
    def sortColors(self, A):
        size = len(A)
        if size<=1:
            return
        record=[0,0,0]
        for i in A:
            record[i]+=1
        record[1] +=record[0]
        record[2] +=record[1]
        for i in range(0,size):
            if i < record[0]:
                A[i]=0
            elif i <record[1]:
                A[i]=1
            else :
                A[i]=2
        return 

File: Cpp/71-80/Sort Colors.cpp
class Solution {
public:
    //count sort
    void sortColors(vector<int>& nums) {
        vector<int > vec(3,0);
        for(int i=0; i<nums.size();i++)
          vec[nums[i]]++;
        int end = 0,i=0;
        for(int j=0;j<3;j++){
          end += vec[j];
          for(; i<end; i++)
            nums[i] = j;
        }
    }
};
